scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/BucketMetadata.java

```diff
@@ -371,6 +393,19 @@ int rehashBucket(byte[] keyBytes, int newNumBuckets) {
   // Serialization
   ////////////////////////////////////////
 
+  private static class HashFunctionSupplier implements Supplier<HashFunction>, Serializable {
```

Comment:
this class/signature is required for Guava's Suppliers::memoize to construct a Serializable memoized supplier: https://github.com/google/guava/blob/v32.1.2/guava/src/com/google/common/base/Suppliers.java#L112-L115